core/common/src/main/java/alluxio/conf/PropertyKey.java

```diff
@@ -1671,9 +1663,17 @@ public String toString() {
           .setConsistencyCheckLevel(ConsistencyCheckLevel.ENFORCE)
           .setScope(Scope.MASTER)
           .build();
+  public static final PropertyKey MASTER_BACKUP_STATE_LOCK_FORCED_DURATION =
+      new Builder(Name.MASTER_BACKUP_STATE_LOCK_FORCED_DURATION)
+          .setDefaultValue("15min")
+          .setDescription("Exclusive locking of the state-lock will timeout after "
+              + "this duration is spent on forced phase.")
+          .setConsistencyCheckLevel(ConsistencyCheckLevel.ENFORCE)
+          .setScope(Scope.MASTER)
+          .build();
   public static final PropertyKey MASTER_BACKUP_SUSPEND_TIMEOUT =
       new Builder(Name.MASTER_BACKUP_SUSPEND_TIMEOUT)
-          .setDefaultValue("1min")
+          .setDefaultValue("3min")
```

Comment:
From my understand of the code, this will describe how long a secondary master will be paused waiting for the primary master to acquire the lock on its journal.
https://github.com/Alluxio/alluxio/blob/2e8258c10319d53bc9e70ee7c7ead04d0b46c999/core/server/master/src/main/java/alluxio/master/backup/BackupLeaderRole.java#L290
https://github.com/Alluxio/alluxio/blob/2e8258c10319d53bc9e70ee7c7ead04d0b46c999/core/server/master/src/main/java/alluxio/master/backup/BackupWorkerRole.java#L180
I would assume that if a primary master cannot acquire a state lock it is because there are a lot of writes to the journal, so in this case wouldn't suspending the secondary master for longer cause it to get farther behind? Otherwise I guess it could be because the master is unable to commit to the journal because of a network partition for example and an operation is holding the lock trying to commit? I guess there are also weird cases where leadership can change in this case and you wouldn't want to suspend the secondary as there might be a new primary committing entries.

So then, I wonder if it is better to suspend the secondary after the lock is taken on the master, instead of before? Maybe then would need to have an extra message step to ensure the master state is at least advanced as the state the secondary is suspended in. I believe that if the primary master cannot advance to the position of the secondary it is contacting, then it would no longer be the primary master.

Reply:
Right. Actually that was the initial design. I can't remember now why we chose to suspend before. I'm aware this flow is not tip-top, so have a ticket to revisit it https://tachyonnexus.atlassian.net/browse/AE-2276